Dockerfile

```diff
@@ -0,0 +1,32 @@
+# Dockerfile
+FROM barichello/godot-ci:4.3
+
+# Set working directory
+WORKDIR /app
+
+RUN apt update -y && apt install -y scons g++
+RUN wget https://github.com/godotengine/godot-cpp/archive/refs/tags/godot-4.3-stable.tar.gz
+RUN tar -xf godot-4.3-stable.tar.gz
+RUN mv godot-cpp-godot-4.3-stable godot-cpp
+
+RUN wget https://github.com/bitwes/Gut/archive/refs/tags/v9.3.0.tar.gz
+RUN tar -xf v9.3.0.tar.gz
+RUN mkdir vendor && mv Gut-9.3.0 vendor/gut
```

Comment:
We should init submodules or not do it there at all.

Submodule points to specific commits that may diverge from the zip file

Do recursive check on checkout GitHub action instead and init submodules using git there

Reply:
True. 


When creating docker image you cannot modify the source, so image build will rely on submodule initialized by the operator, which can be off. I came to the conclusion that atomic and 100% reproducible image build is more important.

In fact, GUT framework added as a submodule is only for adding it to the demo project during development. It can be skipped if all of us will use docker to run tests (but I assume you would like to run them locally?)


> Do recursive check on checkout GitHub action instead and init submodules using git there

Ok, I'll try later, but note that this would be fine for CI. Building on your local machine will require proper manual submodules initialization. As humans we're assuming that docker images will complete and always working, and this requirement could be counter-intuitive.

Reply:
As I said. Docker can init submodules using git command and that will work :)

Reply:
> As I said. Docker can init submodules using git command and that will work :)

True. However, it will not work as expected in a local environment if the submodules are uninitialized or not updated. Currently Dockerfile is fully independent from manual operations, so you can just clone the repo and build docker image without setting up git submodules.

I'm thinking of moving the GUT download to SConstruct and removing the submodule along with wget in the Dockerfile. This will require to write downloader+builder using `urllib` and `zipfile` or `tarfile` (python builtins, so SHOULD be cross platform).